src/orders/service/order-detail.service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { OrderDetailRepository } from '../repository/order-detail.repository';
+import { OrderDetail } from '../domain/order-detail.domain';
+import { StockService } from 'src/products/service/stock.service';
+import { Stock } from 'src/products/domain/stock.domain';
+
+@Injectable()
+export class OrderDetailService {
+  constructor(
+    private readonly orderDetailRepository: OrderDetailRepository,
+    private readonly stockService: StockService,
+  ) {}
+
+  getAllOrderDetails(): Promise<OrderDetail[]> {
+    return this.orderDetailRepository.getAllOrderDetails();
+  }
+
+  getAllOrderDetailsOfOrder(orderId: string): Promise<OrderDetail[]> {
+    return this.orderDetailRepository.getAllOrderDetailsOfOrder(orderId);
+  }
+
+  async getOrderDetailById(
+    orderId: string,
+    productId: string,
+  ): Promise<OrderDetail | null> {
+    const orderDetail: OrderDetail =
+      await this.orderDetailRepository.getOrderDetailById(orderId, productId);
+    if (!orderDetail) {
+      throw new NotFoundException('The order detail was not found');
+    }
+    return orderDetail;
+  }
+
+  async createOrderDetail(orderDetail: OrderDetail): Promise<OrderDetail> {
+    const stock: Stock = await this.stockService.getStockById(
+      orderDetail.productId,
+      orderDetail.shippedFrom.id,
+    );
+
+    await this.stockService.updateStock(
+      orderDetail.productId,
+      orderDetail.shippedFrom.id,
+      { ...stock, quantity: stock.quantity - orderDetail.quantity },
+    );
+    return await this.orderDetailRepository.createOrderDetail(orderDetail);
+  }
+
+  async updateOrderDetail(
```

Comment:
Here we can maybe write less code. Something like this not sure it works exactly as I wrote it:

 `async updateOrderDetail(
    orderId: string,
    productId: string,
    newOrderDetail: OrderDetail,
  ): Promise<OrderDetail> {
    const currentOrderDetail: OrderDetail =
      await this.orderDetailRepository.getOrderDetailById(orderId, productId);
    const stock: Stock = await this.stockService.getStockById(
      currentOrderDetail.productId,
      currentOrderDetail.shippedFrom.id,
    );
    const quantityDifference = newOrderDetail.quantity - currentOrderDetail.quantity;
    if (quantityDifference !== 0) {
      await this.stockService.updateStock(
        currentOrderDetail.productId,
        currentOrderDetail.shippedFrom.id,
        {
          ...stock,
          quantity: stock.quantity - quantityDifference,
        }
      );
    }
    return this.orderDetailRepository.updateOrderDetail(
      orderId,
      productId,
      newOrderDetail,
    );
  }`

Reply:
Not sure why is formatted like this